pymtl3/passes/rtlir/structural/test/StructuralRTLIRGenL2Pass_test.py

```diff
@@ -28,14 +30,14 @@ def construct( s ):
       s.connect( s.out, s.in_.foo )
   a = A()
   a.elaborate()
-  a.apply( StructuralRTLIRGenL2Pass() )
+  a.apply( StructuralRTLIRGenL2Pass( *gen_connections( a ) ) )
```

Comment:
so passing in this as argument is for testing?

Reply:
No. The translators need to pass these parameters in as well. The reason I did this is try to avoid redundant computation of the global connections. For example, if there are 2 components that are marked as "to be translated" in the hierarchy, `StructuralRTLIRGenPass` will be called twice and once on each component. Previously `StructuralRTLIRGenPass` generates all connections from the top component per call. If we pass in the connections (generated elsewhere) we can save these unnecessary computation.